### PR TITLE
feat: add JSON Schema 2020-12 schemas

### LINKED ISSUE #71: JSON Schema 2020-12 schema
I've updated the v1.0 schema

One clarification, the `allOf` for `update` and `remove` scenario
https://github.com/OAI/Overlay-Specification/blob/d029f0d7d2c8861e00de4467ab795e86d9e35824/versions/1.0.0.md?plain=1#L122

> This field has no impact if the `remove` field of this action object is `true`.

Is this something that needs to be enforced with validation?

### PR DESCRIPTION
**Pull request by jeremyfiel**
closes #71

### CODE REVIEW

File: schemas/v1/schema.json
Comment (ralfhandl) [inline schemas/v1/schema.json:1]: I suggest following the folder name convention in https://github.com/OAI/OpenAPI-Specification/tree/main/schemas and rename this folder to `v1.0`

**Comment by jeremyfiel** [comment]
updated the schema to include a validation check for `action-object`. 

> If the `target` selects an object node, the value of this field MUST be an object with the properties and values to merge with the selected node.

---

Because an `array` update can be of `type: any`, I did not include an additional validation check for that scenario outlined as a `MUST`. 
> If the `target` selects an array, the value of this field MUST be an entry to append to the array.

File: schemas/v1.0/schema.json
Comment (ralfhandl) [inline schemas/v1.0/schema.json:1]: I'd remove this file as we will most likely have a workflow for generating it on the fly for publication, similar to
* https://github.com/OAI/OpenAPI-Specification/pull/4139

**Review by ralfhandl — COMMENTED** [review]
In addition to this we'll also need (not necessarily in this PR)
- [ ] tests for the schema, similar to https://github.com/OAI/OpenAPI-Specification/tree/main/tests/v3.1
- [ ] A `schemas/README.MD` file similar to https://github.com/OAI/OpenAPI-Specification/blob/main/schemas/v3.1/README.md, just one folder down directly in `schemas`

**Comment by jeremyfiel** [comment]
related #86

File: schemas/v1.0/readme.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+```bash
+npm install
+npm test
+```
+
+You can also validate a document individually.
+
+```bash
+node scripts/validate.mjs path/to/document/to/validate.yaml
```
Comment (ralfhandl) [inline schemas/v1.0/readme.md:35]: Oops, forgot to copy this 🙄

Comment (ralfhandl) [reply schemas/v1.0/readme.md:35]: Added: thanks for reminding me!

**Pull request closed without merging**